ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -510,6 +502,14 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = do
 
     processLoE curChain (LgrDB.ledgerDbCurrent ledgerDB) (writeTVar cdbLoEFrag) cdbLoE
 
+    loeFrag <- atomically (cross curChain <$> readTVar cdbLoEFrag) <&> \case
```

Comment:
@amesgen Facundo and I noticed this today, it seems like a straightforward mistake I made, do you concur?

Also now it looks like `prcoessLoE` could just return the fragment without storing it in the state...

Reply:
> Also now it looks like `prcoessLoE` could just return the fragment without storing it in the state...

Yes, though this of course only works in case we are sure we don't want to update the `loeFrag` asynchronously.

Reply:
> Facundo and I noticed this today, it seems like a straightforward mistake I made, do you concur?

Indeed, I don't see any reason to not read the LoE fragment *after* it had a chance to be updated in `processLoE` (I think we discussed this on Monday), as this way, we have the chance to potentially select more blocks in case the LoE advanced (eg due to a disconnect by the GDD logic); but it is not immediately clear to me that the previous state had any immediate problems beyond the general concerns about whether this synchronous approach always works (ie is triggered "sufficiently often" by incoming blocks), cf. https://github.com/IntersectMBO/ouroboros-consensus/pull/932#discussion_r1480634036 or https://github.com/IntersectMBO/ouroboros-consensus/pull/878#discussion_r1463161186.